### DISCUSSION TITLE
Initial notes

### DISCUSSION THREAD

**Discussion by greenc-FNAL** [FNALssi/spack-freeze#1] [General]
Enhancement suggestions from an initial examination:
1. Information obtained from an external invocation of `spack find` may already be available using Spack functions to obtain concretized information such as the prefix path and full specstring from the pre-processed `args.specs()`.
1. One could provide either:
   * filename as a destination for the output YAML, or
   * an environment name for direct insertion of the external's information into the environment's configuration
   If neither are specified, the external's information would be inserted into the current environment if set, or output to the screen if not.

**Comment by marcmengel** [comment]
So  I did a mix of things for 2; if there is no environment active, and you do not give it --file, it prints to stdout.
If you have no environment and give it --file, it writes the file.
If you have an environment and give it --file, it writes the file, and adds an include for it to the environment.
If you have an environment and do not give it --file, it picks one based on package hash, writes it, and adds an include for it to the environment.

So if you're in an environment, and spack freeze 3 different packages,  it will write 3 different freeze-hash.yaml files and include them all in the spack.yaml.

**Comment by marcmengel** [comment]
Working on 1.  using spec.traverse() instead of popening spack find --deps...  looks promising.
